cmd/agent/main.go

```diff
@@ -146,24 +145,25 @@ func main() {
 	}
 }
 
-func dialWithTimeout(apiEndpointPtr *string, insecure bool) (*grpc.ClientConn, error) {
+func dialWithTimeout(apiEndpoint string) (*grpc.ClientConn, error) {
 	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Minute)
 	defer cancel()
 
-	// Use TLS by default
-	tlsCredentials := credentials.NewTLS(&tls.Config{
-		MinVersion: tls.VersionTLS13,
-	})
-	transportSecurity := grpc.WithTransportCredentials(tlsCredentials)
+	// insecure by default because we can run it on localhost or in a private network
+	transportSecurity := grpc.WithInsecure()
 
-	// Fall back to plain mode without TLS when explicitly asked to
-	if insecure {
-		transportSecurity = grpc.WithInsecure()
+	// Use TLS if explicitly asked or no schema is in the target
+	if strings.Contains(apiEndpoint, "https://") || !strings.Contains(apiEndpoint, "://") {
```

Comment:
Can we test a few of these assumptions too? E.g. that `https://` and `unix://` endpoint URI's will get secure settings, but `http://` will get insecure settings?

This will also probably result in moving this logic into a separate functions, which will simplify the code a bit.

Reply:
See 461e58efa5c6385a5f8220c7f874cb3b61c4230c

Asserting DialOptions appear to be a pain so I extracted the insecure logic in a function to cover it with tests.